### PR TITLE
DOC: Clarify correlate function definition

### LINKED ISSUE #23782: DOC: Source of "generally defined" function in numpy.correlate needs to be made clear to user
### Issue with current documentation:

In the [Numpy.correlate documentation](https://numpy.org/doc/stable/reference/generated/numpy.correlate.html), we state that the function used for cross-correlating two single dimensional vectors is ' generally defined in signal processing texts':

https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/numeric.py#L682-L686

Furthermore, another formula is also mentioned in the Notes section:
https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/numeric.py#L712-L719

This creates two issues:
1. It is not clear which formula is used by NumPy in the numpy.correlate function
2. To someone less experienced in this signal analysis, this can look confusing. Please note that NumPy (thanks to its low barrier of entry) may be the first exposure to this mathematical expression. 

### Idea or request for content:

The documentation is improved by these two steps:

1.  The formula/algorithm used in numpy.correlate is clarified. If two formula are considered mathematically/computationally equivalent, this is stated clearly 
2. Adding a source for the function mentioned. As an example, we can see this done in [numpy.outer](https://numpy.org/doc/stable/reference/generated/numpy.outer.html).
     - https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/numeric.py#L865
     - https://github.com/numpy/numpy/blob/8cec82012694571156e8d7696307c848a7603b4e/numpy/core/numeric.py#L900-L904

### PR DESCRIPTION
**Pull request by undermyumbrella1**
Fixes #23782

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I'm not sure it adds a ton of value to reference 60+ year old text books that are not open access. Is there any way we could get away with a ref to an open resource like wikipedia instead?

**Comment by charris** [comment]
> reference 60+ year old text books

Hey, I'm 60+ years old :)

**Comment by rossbar** [comment]
> Hey, I'm 60+ years old :)

... but you're also quite available on GitHub :pray: ! My main contention is that there isn't a lot of value in linking to references that others don't have access to.

File: numpy/core/numeric.py [hunk 1]
```diff
@@ -692,17 +692,27 @@ def correlate(a, v, mode='valid'):
...
-    The definition of correlation above is not unique and sometimes correlation
-    may be defined differently. Another common definition is:
+    The definition of correlation above is used to implement the correlate function.
+    This definition is not unique and sometimes correlation
+    may be defined differently. Another common definition is [2]_ :
 
     .. math:: c'_k = \sum_n a_{n} \cdot \overline{v_{n+k}}
 
-    which is related to :math:`c_k` by :math:`c'_k = c_{-k}`.
+    :math:`c_k` is equivalent to :math:`c_{-k}` by :math:`c'_k = c_{-k}`.
```
Comment (aliphys) [inline numpy/core/numeric.py:701]: Mathematically or computationally equivalent? See my comment here: https://github.com/numpy/numpy/issues/23782#issuecomment-1556158295

Comment (ellaella12) [reply numpy/core/numeric.py]: I have updated the documentation to state c'_k is mathematically and computationally equivalent to c_{-k}

File: numpy/core/numeric.py [hunk 2]
```diff
@@ -692,17 +692,24 @@ def correlate(a, v, mode='valid'):
...
+    :math:`c_{-k}` by :math:`c'_k = c_{-k}`.
 
     `numpy.correlate` may perform slowly in large arrays (i.e. n = 1e5) because it does
     not use the FFT to compute the convolution; in that case, `scipy.signal.correlate` might
     be preferable.
 
+    References
+    ----------
+    Wikipedia, "Cross-correlation",
+    https://en.wikipedia.org/wiki/Cross-correlation
```
Comment (rossbar) [inline numpy/core/numeric.py:712]: Per [the numpydoc standard](https://numpydoc.readthedocs.io/en/latest/format.html#references), the `References` section is intended for enumerating reference targets for in-line references in the docstring (i.e. there should be a `.. [1]` and corresponding `[1]_` in the text).

**Review by rossbar — COMMENTED** [review]
I'm not sure these changes do anything to improve clarity. IMO the original docstring was more clear by virtue of it's brevity. That's just my opinion though - @aliphys does this clarify the situation for you?

**Comment by seberg** [comment]
Thanks, adding the references is nice.

**Pull request merged**